superset-frontend/src/components/ListView/types.ts

```diff
@@ -55,11 +55,14 @@ type FilterOperator =
   | 'all_text'
   | 'chart_all_text'
   | 'dataset_is_null_or_empty'
-  | 'between';
+  | 'between'
+  | 'dashboard_is_fav'
+  | 'chart_is_fav';
 
 export interface Filter {
   Header: ReactNode;
   id: string;
+  col?: string;
```

Comment:
This shouldn't be necessary. You can use `id` which corresponds to the column being filtered. If this was added because the `ListView` component throws an error because of a filter on a column that isn't included in the columns, you can add a hidden column entry. 

```ts
{
  accessor: 'id',
  hidden: true,
  disableSortBy: true,
}
```

Reply:
The reason for this is because the backend is expecting the column "id" when filtering by favorite. But that is not very human friendly, as in the URL you will read a filter `filters=(id!t)`. Basically, with this have I added the ability to use a humanly friendly parameter while specifying the actual parameter that is sent to the backend for filtering. I am open to better ideas though.

Reply:
Ok. Rereading your comment it looks like I could actually achieve the same result with your approach. I will try it. Thanks for the hint.

Reply:
hmm, I guess the main issue here is that the api for this filter is not very human readable. So far these filters have correspond to their api counterparts. Maybe adding the the operators to the url would make it less confusing? Something like `filters=(id: !(chart_is_fav,!t)` could work, though that adds a bit of bloat.    

I can see how having the option to change the way the filter displays in the url would be useful. Could we instead change it to `urlDisplay` or `urlKey` or something along those line. That way we'd only have to change this line
https://github.com/apache/incubator-superset/blob/7dac150dc6f9279c93a832e509aaa89bd35412bb/superset-frontend/src/components/ListView/utils.ts#L297 and this function
https://github.com/apache/incubator-superset/blob/7dac150dc6f9279c93a832e509aaa89bd35412bb/superset-frontend/src/components/ListView/utils.ts#L126